frontend/catalyst/utils/gen_mlir.py

```diff
@@ -50,14 +50,17 @@ def gen_teardown(ctx):
     return ir.Module.parse(txt, ctx)
 
 
-def inject_functions(module, ctx):
+def inject_functions(module, ctx, seed):
     """
     This function appends functions to the input module.
     """
     # Add C interface for the quantum function.
     module.body.operations[0].attributes["llvm.emit_c_interface"] = ir.UnitAttr.get(context=ctx)
 
     setup_module = gen_setup(ctx)
+    setup_module.body.operations[0].body.blocks[0].operations[0].attributes["seed"] = (
+        ir.StringAttr.get(seed, context=ctx)
+    )
```

Comment:
You can even write the MLIR directly in the setup function.

Reply:
I was trying to make sure the attribute is indeed a string attribute, but I guess it doesn't matter since IRs can even be manually written. Putting this in the text above directly instead of going through this parade of accesses would indeed be clearer.